### PR TITLE
New Message and Error for Delete Profile

### PR DESCRIPTION
**Pull request by prachigarg19**
Displays error  if user tries to delete their account with active subscriptions.

Changes in this pull request:
- Display message to make sure user have cancelled all subscriptions before delete button.
- Create modal which pops up when user continues deleting their account with active subscriptions: 
  If continue is clicked, the user is redirected to profile/recurrency. 
- After cancelling all subscriptions, user can delete the account successfully.

### CODE REVIEW

**Comment by mariahosfeld** [comment]
- [ ] Please change the wording of "Before proceeding, make sure you've deleted all subscriptions before ." to "Before proceeding, make sure you've cancelled all subscriptions."

**Comment by mariahosfeld** [comment]
The basic functionality works, but please make the error message when a Subscription is still there more prominent, e.g. in red and bold type, so that the user will spot it immediately.

**Review by norbertschuler — COMMENTED** [review]
LGFM, first I was puzzled if "continue" means continue to delete my profile anyway, but then reading the text I understood continue to be redirected to the subscriptions to be able to cancel them. Maybe the button could be renamed to "show my subscriptions"?

**Comment by mariahosfeld** [comment]
Functionality works as expected for me.

File: src/features/common/Layout/CustomModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          <div className={styles.modal__subtitle}>{modalSubtitle}</div>
+        </div>
+        <div
+          className={styles.buttonContainer}
+          style={{ justifyContent: 'center' }}
+        >
+          <button
+            id={'Continue'}
+            className={styles.continueButton}
+            onClick={onClick1}
```
Comment (Shreyaschorge) [inline src/features/common/Layout/CustomModal/index.tsx:43]: @prachigarg19 Please give it a meaningful name, maybe something like `handleContinue`

File: src/features/user/Settings/DeleteProfile.tsx [hunk 2]
```diff
@@ -18,25 +19,36 @@ export default function DeleteProfile({}: any) {
...
   const [canDeleteAccount, setcanDeleteAccount] = React.useState(false);
 
   const handleDeleteAccount = () => {
     setIsUploadingData(true);
     deleteAuthenticatedRequest('/app/profile', token, handleError).then(
       (res) => {
-        if (res !== 404) {
-          logoutUser(`${process.env.NEXTAUTH_URL}/`);
-        } else {
+        if (res == 400) {
```
Comment (Shreyaschorge) [inline src/features/user/Settings/DeleteProfile.tsx:29]: @prachigarg19 There could be multiple factors that might result into a 400 response. Before opening the Cancel Subscriptions modal, along with 400 response please check the response (i.e error_code, error_type) returned from the server. 
image

File: src/features/common/Layout/CustomModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          >
+            {buttonTitle}
+          </button>
+          <button
+            id={'Cancel'}
+            className={styles.cancelButton}
+            onClick={onClick2}
```
Comment (Shreyaschorge) [inline src/features/common/Layout/CustomModal/index.tsx:50]: @prachigarg19 Please give it a meaningful name, maybe something like `handleCancel`

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
Functionally it works. There are only minor changes that needs to be handled.

**Review by Shreyaschorge — APPROVED** [review]
LGFM. Only thing is we need to manage our errors in a better way.

**Pull request merged**